### PR TITLE
Refactor command handlers

### PR DESCRIPTION
**Pull request by ImprobableNic**
#### Description
A bit of refactoring around the various spatial command handlers, which IMO were unnecessarily scattered between different object, and using delegates where simple TFunction could be used.

### CODE REVIEW

**Review by samiwh — APPROVED** [review]
Discussed offline. I agree with the TFunction changes. The main question is whether or not the somewhat general handler is the right way to go or if we should be making a generic handler for all commands. Or if most systems should be doing this more manually.

However this makes the code currently there easier to read and the above follow up questions are still easier to deal with if we merge this in.

File: SpatialGDK/Source/SpatialGDK/Public/Interop/ClaimPartitionHandler.h [hunk 1]
```diff
@@ -5,22 +5,27 @@
...
-	void ClaimPartition(Worker_EntityId SystemEntityId, Worker_PartitionId PartitionToClaim);
+	void ClaimPartition(ISpatialOSWorker& WorkerInterface, Worker_EntityId SystemEntityId, Worker_PartitionId PartitionToClaim);
+	void ClaimPartition(ISpatialOSWorker& WorkerInterface, Worker_EntityId SystemEntityId, Worker_PartitionId PartitionToClaim,
+						SystemEntityCommandDelegate Delegate);
 
 	void ProcessOps(const TArray<Worker_Op>& Ops);
 
 private:
-	TMap<Worker_RequestId_Key, Worker_PartitionId> ClaimPartitionRequestIds;
+	struct ClaimPartitionRequest
```
Comment (improbable-dmitrii) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/ClaimPartitionHandler.h:23]: `FClaimPartitionRequest`?

File: SpatialGDK/Source/SpatialGDK/Public/Interop/EntityQueryHandler.h [hunk 2]
```diff
@@ -13,9 +13,11 @@
 
 DECLARE_CYCLE_STAT(TEXT("EntityQueryHandler"), STAT_EntityQueryHandler, STATGROUP_SpatialNet);
 
+using EntityQueryDelegate = TFunction<void(const Worker_EntityQueryResponseOp&)>;
```
Comment (improbable-dmitrii) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/EntityQueryHandler.h:16]:
```suggestion
using FEntityQueryDelegate = TFunction<void(const Worker_EntityQueryResponseOp&)>;
```

**Review by improbable-dmitrii — APPROVED** [review]
The only thing I can note is that maybe it'd be better to add `F` prefixes to all types, not only the command handlers themselves?

**Pull request merged**